compiler/src/main/java/com/dylibso/chicory/compiler/Cache.java

```diff
@@ -0,0 +1,37 @@
+package com.dylibso.chicory.compiler;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public interface Cache {
+
+    /**
+     * The temp dir is deleted once closed.
+     */
+    interface TempDir extends Closeable {
```

Comment:
This is a "marker interface" I'd prefer to hide the complexity rising the API abstraction other than exposing this.